pkg/pipeline/repository/PipelineStageRepository.go

```diff
@@ -230,30 +261,52 @@ func (impl *PipelineStageRepositoryImpl) GetCiStageByCiPipelineIdAndStageType(ci
 	return &pipelineStage, nil
 }
 
-func (impl *PipelineStageRepositoryImpl) CreateCiStage(ciStage *PipelineStage) (*PipelineStage, error) {
-	err := impl.dbConnection.Insert(ciStage)
+func (impl *PipelineStageRepositoryImpl) GetCdStageByCdPipelineIdAndStageType(cdPipelineId int, stageType PipelineStageType) (*PipelineStage, error) {
+	var pipelineStage PipelineStage
+	err := impl.dbConnection.Model(&pipelineStage).
+		Where("cd_pipeline_id = ?", cdPipelineId).
+		Where("type = ?", stageType).
+		Where("deleted = ?", false).Select()
 	if err != nil {
-		impl.logger.Errorw("error in creating pre stage entry", "err", err, "ciStage", ciStage)
+		impl.logger.Errorw("err in getting cd stage by cdPipelineId", "err", err, "cdPipelineId", cdPipelineId)
 		return nil, err
 	}
-	return ciStage, nil
+	return &pipelineStage, nil
 }
 
-func (impl *PipelineStageRepositoryImpl) UpdateCiStage(ciStage *PipelineStage) (*PipelineStage, error) {
-	err := impl.dbConnection.Update(ciStage)
+func (impl *PipelineStageRepositoryImpl) CreatePipelineStage(pipelineStage *PipelineStage, tx *pg.Tx) (*PipelineStage, error) {
+	if tx != nil {
+		err := tx.Insert(pipelineStage)
+		if err != nil {
```

Comment:
we can merge error message of both the operations

Reply:
done